source/developers-guide/concept-cart-bundle/index.md

```diff
@@ -0,0 +1,424 @@
+        echo "\n tax amount of delivery: " . $price->getCalculatedTaxes()->getAmount();
+
+        echo "\n\n address: ";
+        echo ' ' . $delivery->getAddress()->getFirstname();
+        echo ' ' . $delivery->getAddress()->getLastname();
+        echo ' ' . $delivery->getAddress()->getStreet();
+        echo ' ' . $delivery->getAddress()->getZipcode();
+        echo ' ' . $delivery->getAddress()->getCity();
+
+        echo "\n delivery date: ";
+        echo $delivery->getDeliveryDate()->getEarliest()->format('Y-m-d');
+        echo ' - ';
+        echo $delivery->getDeliveryDate()->getLatest()->format('Y-m-d');
+
+        /** @var CartBundle\Domain\Delivery\DeliveryPosition $position */
+        foreach ($delivery->getPositions() as $i => $position) {
+            echo "\n\n position " . $i;
+            echo "\n quantity " . $position->getQuantity();
+            echo "\n price in delivery: " . $position->getPrice()->getPrice();
+        }
+
+    }
+}
+```
+
+## Architecture
+
+### Cart layers
+Inside the calculation process, the cart passes through different states. To provide a valid state for each service layer the this states are reflected in different classes:
+* `CartBundle\Domain\Cart\Cart`
+    * Defines which line items has to be calculated inside the process
+* `CartBundle\Domain\Cart\ProcessorCart`
+    * Defines which line items already calculated and which deliveries already generated 
+* `CartBundle\Domain\Cart\CalculatedCart`
+    * Contains a list of all calculated line items 
+    * Contains a collection of all generated deliveries
+    * Has a calculated price with summed taxes, tax rules and net or gross prices 
+
+### Processor concept 
+The following diagram shows the architecture behind the cart process for product calculation:   
+
+![image](img/img.001.png)
+
+The cart calculation are done over the `CartBundle\Domain\Cart\CartCalculator` class. 
+This class contains a list of `CartBundle\Domain\Cart\CartProcessorInterface` which are the access points for the shopware core and third party developers in the cart process. 
+```php
+interface CartProcessorInterface
+{
+    public function process(
+        CartBundle\Domain\Cart\Cart $cart,
+        CartBundle\Domain\Cart\ProcessorCart $processorCart,
+        CartContextInterface $context
+    );
+}
+
+```
+
+### Domain and Infrastructure layers
+We have taken great care in the design of the architecture, to separate calculation process and shopware access.
+That means all shopware specify operations like database access, delivery information or price selections (with graduated prices or price groups) are separated into own gateways which can be replaced with other data sources.
+This layers are named **Domain** and **Infrastructure** and placed on the first level of the `CartBundle`.
+All classes inside the domain layer have, with small exclusions (shop, customer group, currency), a dependency on the shopware core or database.
```

Comment:
The domain layer should not have any dependencies to the shopware core. That's the infrastructure/bundle layer.